src/Dstarlite.js

```diff
@@ -0,0 +1,115 @@
+import Heap from './Heap';
+import Cell from './Cell';
+
+export default class Dstarlite {
+    constructor(start, goal, heuristic) {
+        this.Heap = new Heap();
+        this.km = 0;
+        this.rhs = new Map();
+        this.gscore = new Map();
+        this.heuristic = heuristic;
+        this.start = start;
+        this.goal = goal;
+    }
+
+    initialize() {
+        const grid = this.goal.grid.grid;
```

Comment:
i really really really want to get rid of all direct accesses to properties on `Grid`

can we make an iterator that just goes over all the cells? you will have to look this up and/or ask me for more deets

Reply:
🐇